### ISSUE TITLE
alertmanager crd needs support for invalid enum names

### ISSUE THREAD

**Issue by clux** [kube-rs/kopium#92]
In their schema:

```yaml
                          matchType:
                            description: Match operation available with AlertManager
                              >= v0.22.0 and takes precedence over Regex (deprecated)
                              if non-empty.
                            enum:
                            - '!='
                            - =
                            - =~
                            - '!~'
                            type: string
```

This forces us to basically come up with our own names for the enum variants (because we can't simply drop invalid characters because they are all invalid in identifiers - even if wrapped in a rawstring).

So we have an immediate first step:

- implement a `Container::invalid_enum_identifiers(&self) -> bool` that checks if any of the members have such a name in the enum
- call that function in main, and if it's true; use generic `Variant0`, `Variant1` names for them (but with `#[serde(rename)]` attrs to keep it compliant)

..and as a potential third step; args/config for letting users customize the values of these Variant names since it's pretty unergonomic to work with. First priority is to get it to compile though.

**Comment by clux** [comment]
The function that detects validity could maybe use a regex or maybe https://docs.rs/syn/1.0.103/syn/fn.parse_str.html

**Comment by Dav1dde** [comment]
Fixed with #103
